play-json/src/main/scala/play/api/libs/json/JsConstraints.scala

```diff
@@ -16,9 +16,12 @@ trait PathFormat {
   def at[A](path: JsPath)(implicit f: Format[A]): OFormat[A] =
     OFormat[A](Reads.at(path)(f), Writes.at(path)(f))
 
-  def nullable[A](path: JsPath)(implicit f: Format[A]): OFormat[Option[A]] =
+   def nullable[A](path: JsPath)(implicit f: Format[A]): OFormat[Option[A]] =
     OFormat(Reads.nullable(path)(f), Writes.nullable(path)(f))
 
+  def optional[A](path: JsPath, defaultValue: => Option[A])(implicit f: Format[A]): OFormat[Option[A]] =
```

Comment:
Well, there are two distinct cases.
We could define default value for generic field as well as define it for optional one.
Why so.. Handling reads on optional fields with default values make us think about how could we say reset optional value if there if some default.
```scala
case class Foo(bar: Option[String] = Some("bar"))
```
So. if there is no `bar` field in the payload - according to semantic we have to use default value, but is there any way to set it to None? That's a question and that's why you see this construction.